### PR TITLE
Data table styling

### PR DESCRIPTION
**Pull request by SasithaRajapakasha**
**Changes**
Style update to data table

- [delete] branch

**Additional information**

- You may refer to this [BOOKINGSG-4463](https://jira.ship.gov.sg/browse/BOOKINGSG-4633)

### CODE REVIEW

File: src/data-table/data-table.tsx [hunk 1]
```diff
@@ -124,10 +131,11 @@ export const DataTable = ({
             <HeaderCell
                 data-testid={getDataTestId("header-selection")}
                 $clickable={false}
+                $isCheckbox={true}
             >
                 <CheckBoxWrapper>
                     {enableSelectAll && (
-                        <CheckboxElement
+                        <Checkbox
```
Comment (qroll) [inline src/data-table/data-table.tsx:138]: there should be an indeterminate state set for the case when not all items are selected

**Review by qroll — COMMENTED** [review]
accessibility note: Voiceover does not announce the column header but the table markup seems fine, will try to find out why

**Comment by qroll** [comment]
a scrollbar appears for the loading state

Screenshot 2023-09-05 at 1 48 55 PM

File: src/data-table/data-table.tsx (hunk 1 above)
Comment (SasithaRajapakasha) [reply src/data-table/data-table.tsx:138]: I have updated indeterminate changes with next changes, Can I do the update of the indeterminate in there?

File: src/data-table/data-table.styles.tsx [hunk 2]
```diff
@@ -35,7 +35,6 @@ export const TableWrapper = styled.div`
     width: 100%;
     border: 1px solid ${borderColor};
     border-radius: 0.5rem;
-    overflow: auto;
```
Comment (qroll) [inline src/data-table/data-table.styles.tsx:27]: this is needed to constrain the table backgrounds within the border radius, so you'll have to put this back

as for the scrollbar, I found that an alternative fix is to set `overflow: hidden;` on the `LoaderWrapper`. can add a comment in the code if you decide to use this fix

**Comment by qroll** [comment]
> accessibility note: Voiceover does not announce the column header but the table markup seems fine, will try to find out why

Hmm I tried again on a different Macbook and it's working as expected. I assume it's a bug specific to the previous device I was testing on. No action needed.

File: src/data-table/data-table.tsx [hunk 3]
```diff
@@ -180,17 +188,25 @@ export const DataTable = ({
                 data-testid={getDataTestId(`row-${cellId}`)}
                 key={cellId}
                 style={style}
+                $isCheckbox={false}
             >
-                {cellData}
+                {typeof cellData === "string" ? (
+                    <BodyCellContent>{cellData}</BodyCellContent>
+                ) : (
+                    cellData
```
Comment (qroll) [inline src/data-table/data-table.tsx:196]: something that I missed out, if `cellData` is a function, it should be called with the row item and `RowRenderProps`

**Review by qroll — APPROVED** [review]

**Pull request merged**